astropy/cosmology/parameter/tests/test_parameter.py

```diff
@@ -86,20 +85,14 @@ class ParameterTestMixin:
     @pytest.fixture
     def parameter(self, cosmo_cls):
         """Cosmological Parameters"""
-        # I wish this would work
-        # yield from {getattr(cosmo_cls, n) for n in cosmo_cls.__parameters__}
-
-        # just return one parameter at random
-        yield getattr(cosmo_cls, set(cosmo_cls.__parameters__).pop())
+        yield from cosmo_cls.parameters.values()
 
     @pytest.fixture
     def all_parameter(self, cosmo_cls):
         """Cosmological All Parameter instances"""
-        # I wish this would work
-        # yield from {getattr(cosmo_cls, n) for n in cosmo_cls.__all_parameters__}
-
         # just return one parameter at random
-        yield getattr(cosmo_cls, set(cosmo_cls.__all_parameters__).pop())
+        ps = cosmo_cls.parameters | cosmo_cls.derived_parameters
+        yield ps[set(ps).pop()]
```

Comment:
we might get a derived parameter...

Reply:
`set.pop()` returns an arbitrary element from the set and an arbitrary element from `cosmo_cls.parameters` is also a valid arbitrary element from `cosmo_cls.parameters | cosmo_cls.derived_parameters`. This means we can save a line of code and some CPU time by not involving derived parameters here at all.

If you are trying to implement [property-based tests](https://docs.astropy.org/en/latest/development/testguide.html#property-based-tests) you should implement them properly using `hypothesis`.

Reply:
It would be faster to do 
```suggestion
        n = set(cosmo_cls._all_parameters).pop()
        yield self.parameters[n] if n in self.parameters else self.derived_parameters[n]
```

Reply:
I did mention performance, but I'm much more worried that the way this fixture is written is introducing randomness to the tests in an uncontrolled way. If a parameter does cause some test to fail then how simple would it be to replicate that failure? [`hypothesis` remembers parameters that caused test failures](https://hypothesis.readthedocs.io/en/latest/database.html), this fixture does not.

Reply:
Agreed. But this test has been like this for a while now. Worth fixing, but not here.